### ISSUE TITLE
PgAuditLogToFile Simultaneously Creates Two Different Log Files 

### ISSUE THREAD

**Issue by muhabbad** [fmbiete/pgauditlogtofile#35]
I want pgauditlogtofile to automatically produce a new log file with that date and time name every hour. So far, no problem. The problem is that pgauditlogtofile produces a file with a date from last year and keeps this file permanently. If I delete this file, it is automatically recreated. Why does it create this file? It shouldn't create it. It only needs to open the files according to the date and time format in the criteria I gave. 

Here is the config that i use: 

![resim](https://github.com/fmbiete/pgauditlogtofile/assets/126278907/68d2bc11-13fc-4f79-8f8d-0bc9e5707af0)

@fmbiete @nvthongswansea Can you please have a look at my situation? I wish your help.

Thank you.

**Comment by philippe-vs** [comment]
Hi, 

We are facing a similar issue with PostgreSQL 14.11, on a Oracle Linux Server release 9.3, with pgaudit 1.6.2 and pgauditlogtofile 1.5.12 from PGDG repos.

A first audit file is created, and then, for each next days, the first file is appended, plus a new file is created.

Here the configuration parameters used :

```
            name            |             setting             
----------------------------+---------------------------------
 pgaudit.log                | none
 pgaudit.log_catalog        | off
 pgaudit.log_client         | off
 pgaudit.log_connections    | off
 pgaudit.log_directory      | /.../auditlog
 pgaudit.log_disconnections | off
 pgaudit.log_filename       | audit-%Y%m%d_%H%M.log
 pgaudit.log_level          | log
 pgaudit.log_parameter      | off
 pgaudit.log_relation       | off
 pgaudit.log_rotation_age   | 1440
 pgaudit.log_rows           | off
 pgaudit.log_statement      | on
 pgaudit.log_statement_once | off
 pgaudit.role               | auditor
```

I've not identified which PostgreSQL process is appending to the first file, but it gets recreated, even if we delete it.

```
-rw------- 1 postgres postgres  14K 10 avril 11:15 audit-20240326_0000.log
-rw------- 1 postgres postgres  47M 27 mars  14:50 audit-20240327_0000.log
-rw------- 1 postgres postgres 7,8K 28 mars  18:16 audit-20240328_0000.log
-rw------- 1 postgres postgres 4,7K 29 mars  15:21 audit-20240329_0000.log
-rw------- 1 postgres postgres 6,8K  2 avril 16:29 audit-20240402_0000.log
-rw------- 1 postgres postgres 9,9K  3 avril 16:49 audit-20240403_0000.log
-rw------- 1 postgres postgres 5,0K  4 avril 09:09 audit-20240404_0000.log
-rw------- 1 postgres postgres  19K  5 avril 14:49 audit-20240405_0000.log
-rw------- 1 postgres postgres  13K  9 avril 17:35 audit-20240409_0000.log
-rw------- 1 postgres postgres  12K 10 avril 11:15 audit-20240410_0000.log
```

**Comment by halvaborsch** [comment]
Looks like I'm catching the effects of this issue too. 
My case is to create a new log file every two hours with the hour number. After 4 hours old files are deleted by crond. 

```
pgaudit.log_filename = 'pgaudit-%I.log'
pgaudit.log_rotation_age = '120'

```
But still

But all the time transactions continue to get into the file with the name pgaudit-10.log. There are no long open transactions with the database. If you look in _lsof_ process postgres does not keep this file open, only the required hour.  

```
 pgaudit-02.log                                                          │615723K│Apr 22 04:00│
 pgaudit-04.log                                                          │  1719M│Apr 22 06:00│
 pgaudit-06.log                                                          │  8603M│Apr 22 06:43│
 pgaudit-10.log                                                          │  29835│Apr 22 06:40│
```

**Comment by fmbiete** [comment]
Take a look at branch _next_. We are using a background worker to calculate the file to use, which should fix your issue.
There is also a new experimental option to automatically close an audit log file after some inactivity period.

**Comment by fmbiete** [comment]
Solved in v1.6.0
